test/jdk/java/awt/regtesthelpers/PassFailJFrame.java

```diff
@@ -258,30 +262,112 @@ private static void getFailureReason() {
 
     /**
      * Position the instruction frame with testWindow (testcase created
-     * window) by the specified position.
+     * window) by the specified position. If testWindow is null, only
+     * the instruction frame is positioned according to the Position
+     * parameter.
      * Note: This method should be invoked from the method that creates
-     * testWindow.
+     * testWindow. At test-level, the testWindow must be made visible
+     * only after calling this method.
+     *
+     * @param testWindow test window that the test is created. It can be null.
+     * @param position  position can either be:
+     *                  HORIZONTAL - the test instruction frame is positioned
+     *                  such that its right edge aligns with screen's vertical
+     *                  center and the test window (if not null) is placed to
+     *                  the right of the instruction frame.
+     *
```

Comment:
"vertical center" means half way between the top and the bottom which is absolutely not what you mean, is it ?

horizontal center surely ?

But what if the test window is > 0.5 the width of the screen ? Now it is off the edge .. wouldn't it be better
to move the instruction window to the left to accommodate it ?
Although that might be tricky if we don't know its size yet ..

Reply:
> But what if the test window is > 0.5 the width of the screen ? Now it is off the edge .. wouldn't it be better to move the instruction window to the left to accommodate it ?

It would be better. The same problem exists for vertical and horizontal positioning: if the test window is larger than half the width of the screen, its part may become off the edge.

Should this be dealt with in a separate issue?

Reply:
@prrace @aivanov-jdk For Horizontal position, the frame and window are positioned as follows - 

![image](https://user-images.githubusercontent.com/95945681/185998431-93adcced-cef9-4c04-bd2a-4511b6f34210.png). 

I intended to convey that the right edge of the instruction frame aligns with the yellow line, the line that runs vertically and divides the screen into right and left halves (vertical center - probably not the right term, as Phil mentioned horizontal center might be correct word here. Updated it in the latest commit.


> But what if the test window is > 0.5 the width of the screen ? Now it is off the edge .. wouldn't it be better
> to move the instruction window to the left to accommodate it ?
> Although that might be tricky if we don't know its size yet ..

Yes I agree, can we take it up as a separate issue while redesigning the framework as @aivanov-jdk suggested?

Reply:
Yes, a pre-existing issue. Dealing with it later is OK even though it'll slightly respecify the method again.